shared/actions/config.js

```diff
@@ -196,12 +198,97 @@ function _bootstrapSuccess(action: ConfigGen.BootstrapSuccessPayload, state: Typ
   return Saga.all(actions)
 }
 
+function _pgpSecurityModelChangeMessageSaga() {
+  RPCTypes.pgpPgpStorageDismissRpcPromise().catch(err => {
+    console.warn('Error in sending pgpPgpStorageDismissRpc:', err)
+  })
+}
+
+function _loadAvatarHelper(action: {payload: {names: Array<string>, endpoint: string, key: string}}) {
+  const {names, endpoint, key} = action.payload
+  return Saga.all([
+    Saga.call(RPCTypes.apiserverGetRpcPromise, {
+      args: [{key, value: names.join(',')}, {key: 'formats', value: 'square_360,square_200,square_40'}],
+      endpoint,
+    }),
+    Promise.resolve(names),
+  ])
+}
+
+function _afterLoadAvatarHelper([response: {body: string}, names]) {
+  const nameToUrlMap = JSON.parse(response.body).pictures.reduce((map, picMap, idx) => {
+    const name = names[idx]
+    const urlMap = {
+      ...(picMap['square_200'] ? {'200': picMap['square_200']} : null),
+      ...(picMap['square_360'] ? {'360': picMap['square_360']} : null),
+      ...(picMap['square_40'] ? {'40': picMap['square_40']} : null),
+    }
+    map[name] = urlMap
+    return map
+  }, {})
+
+  return Saga.put(ConfigGen.createLoadedAvatars({nameToUrlMap}))
+}
+
+let _avatarsToLoad = {}
+function* _loadAvatars(action: ConfigGen.LoadAvatarsPayload) {
+  const {usernames} = action.payload
+  // store it and wait, once our timer is up we pull any and run it
+  usernames.forEach(username => {
+    _avatarsToLoad[username] = true
+  })
+  yield Saga.call(Saga.delay, 200)
+
+  const names = Object.keys(_avatarsToLoad)
+  _avatarsToLoad = {}
+
+  if (names.length) {
+    yield Saga.put({
+      payload: {endpoint: 'image/username_pic_lookups', key: 'usernames', names},
+      type: '_loadAvatarHelper',
+    })
+  }
+}
+
+let _teamAvatarsToLoad = {}
+function* _loadTeamAvatars(action: ConfigGen.LoadTeamAvatarsPayload) {
+  const {teamnames} = action.payload
+  teamnames.forEach(teamname => {
+    _teamAvatarsToLoad[teamname] = true
+  })
+  // store it and wait, once our timer is up we pull any and run it
+  yield Saga.call(Saga.delay, 200)
+
+  const names = Object.keys(_teamAvatarsToLoad)
+  _teamAvatarsToLoad = {}
+
+  if (names.length) {
+    yield Saga.put({
+      payload: {endpoint: 'image/team_avatar_lookups', key: 'team_names', names},
+      type: '_loadAvatarHelper',
+    })
+  }
+}
+
+// Every minute we clear out any avatars that might have errored out
+function* _periodicAvatarCacheClear(): Generator<any, void, any> {
```

Comment:
we now periodically go through and cleanup missing avatars incase those errors could resolve themselves